Framework/PythonInterface/mantid/utils/absorptioncorrutils.py

```diff
@@ -32,174 +33,245 @@ def _getBasename(filename):
+    return mantid.kernel.ConfigService.getInstrument(ws.getInstrument().getName()).shortName()
 
-    return fileName(full path), sha1
-    """
 
-    # parse algorithm used for absorption calculation
-    # NOTE: the actual algorithm used depends on selected abs_method, therefore
-    #       we are embedding the algorithm name into the SHA1 so that we can
-    #       differentiate them for caching purpose
-    alg_name = {
-        "SampleOnly": "AbsorptionCorrection",
-        "SampleAndContainer": "AbsorptionCorrection",
-        "FullPaalmanPings": "PaalmanPingsAbsorptionCorrection",
-    }[abs_method]
-
-    # key property to generate the HASH
-    if not mtd.doesExist(wkspname):
-        raise RuntimeError(
-            "Could not find workspace '{}' in ADS to create cache name".format(wkspname))
-    ws = mtd[wkspname]
-    # NOTE:
-    #  - The query for height is tied to a beamline, which is not a good design as it
-    #    will break for other beamlines
-    property_string = [
-        f"{key}={val}" for key, val in {
-            'wavelength_min': ws.readX(0).min(),
-            'wavelength_max': ws.readX(0).max(),
-            "num_wl_bins": len(ws.readX(0)) - 1,
-            "sample_formula": ws.run()['SampleFormula'].lastValue().strip(),
-            "mass_density": ws.run()['SampleDensity'].lastValue(),
-            "height_unit": ws.run()['BL11A:CS:ITEMS:HeightInContainerUnits'].lastValue(),
-            "height": ws.run()['BL11A:CS:ITEMS:HeightInContainer'].lastValue(),
-            "sample_container": ws.run()['SampleContainer'].lastValue().replace(" ", ""),
-            "algorithm_used": alg_name,
-        }.items()
-    ]
-
-    cache_path, signature = mantid.simpleapi.CreateCacheFilename(
-        Prefix=wksp_prefix,
-        OtherProperties=property_string,
-        CacheDir=cache_dir,
-    )
-
-    return cache_path, signature
-
-
-def _getCachedData(absName, abs_method, sha1, cache_file_name):
-    """
-    Check both memory and disk to locate cache data.  Returns ("", "")
-    if no cache can be found.
+def __get_cache_name(meta_wksp_name, abs_method, cache_dirs=[], prefix_name=""):
+    """generate cachefile name (full path) and sha1
 
-    :params absName: absorption workspace name base
-    :params abs_method: absorption calculation method
-    :params sha1: SHA1 that identify cached workspace
-    :params cache_file_name: cache file name to search
+    :param meta_wksp_name: name of workspace contains relevant meta data for hashing
+    :param abs_method: method used to perform the absorption calculation
+    :param cache_dir: cache directory to scan/load cache data
+    :param prefix_name: prefix to add to wkspname for caching
 
-    return cached_absorption_correction_workspace_name_sample,
-           cached_absorption_correction_workspace_name_container
+    return cachefile_name: full path of the cache file
+           sha1: MD5 value based on selected property
```

Comment:
please update docstring after changing the function signature

Reply:
done